Build on FreeBSD 12.2-RELEASE-p3 is broken

Compile log: http://paste.debian.net/plainh/5d93a9c7

Other details:
```
[nico@hades ~/src/bm]$ uname -apKU
FreeBSD hades.herrhotzenplotz.geek 12.2-RELEASE-p3 FreeBSD 12.2-RELEASE-p3 GENERIC  amd64 amd64 1202000 1202000
[nico@hades ~/src/bm]$ cc --version
FreeBSD clang version 10.0.1 (git@github.com:llvm/llvm-project.git llvmorg-10.0.1-0-gef32c611aa2)
Target: x86_64-unknown-freebsd12.2
Thread model: posix
InstalledDir: /usr/bin
[nico@hades ~/src/bm]$ 
```

I have no idea what is going on with the static assert, but the stat errors seem to be related to preprocessor magic before the includes. See: https://www.freebsd.org/cgi/man.cgi?query=stat&apropos=0&sektion=2&manpath=FreeBSD+12.2-RELEASE+and+Ports&arch=default&format=html

EDIT:
It does compile, if I comment out on the top of `basm.c`:

```
//#    define _DEFAULT_SOURCE
//#    define _POSIX_C_SOURCE 200112L
```

Interestingly enough, it doesn't even compile in a Debian Linuxulator environment:

```
[nico@hades ~/src/bm]$ gcc --version
gcc (Debian 6.3.0-18+deb9u1) 6.3.0 20170516
Copyright (C) 2016 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

[nico@hades ~/src/bm]$ uname -a
Linux hades.herrhotzenplotz.geek 3.2.0 FreeBSD 12.2-RELEASE-p3 GENERIC x86_64 GNU/Linux
[nico@hades ~/src/bm]$ cc nobuild.c -o build_looniks
[nico@hades ~/src/bm]$ gcc nobuild.c -o build_looniks
[nico@hades ~/src/bm]$ ./build_looniks tools
[INFO] mkdirs build/
[WARN] directory build/ already exists
[INFO] mkdirs build/library
[WARN] directory build/library already exists
[INFO] cc -Wall -Wextra -Wswitch-enum -Wmissing-prototypes -Wconversion -Wno-missing-braces -pedantic -fno-strict-aliasing -ggdb -std=c11 -c src/library/bm.c -o build/library/bm.o
[INFO] cc -Wall -Wextra -Wswitch-enum -Wmissing-prototypes -Wconversion -Wno-missing-braces -pedantic -fno-strict-aliasing -ggdb -std=c11 -c src/library/arena.c -o build/library/arena.o
[INFO] cc -Wall -Wextra -Wswitch-enum -Wmissing-prototypes -Wconversion -Wno-missing-braces -pedantic -fno-strict-aliasing -ggdb -std=c11 -c src/library/basm.c -o build/library/basm.o
src/library/basm.c: In function ‘path_file_exist’:
src/library/basm.c:593:32: error: ‘S_IFMT’ undeclared (first use in this function)
            ((statbuf.st_mode & S_IFMT) == S_IFREG);
                                ^~~~~~
src/library/basm.c:593:32: note: each undeclared identifier is reported only once for each function it appears in
src/library/basm.c:593:43: error: ‘S_IFREG’ undeclared (first use in this function)
            ((statbuf.st_mode & S_IFMT) == S_IFREG);
                                           ^~~~~~~
src/library/basm.c:600:1: warning: control reaches end of non-void function [-Wreturn-type]
 }
 ^
[ERRO] command exited with exit code 1
[nico@hades ~/src/bm]$ 
```

which uses glibc:

```
[nico@hades ~/src/bm]$ ldd build_looniks 
        linux_vdso.so.1 (0x00007ffffffff000)
        libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x0000000801800000)
        /lib64/ld-linux-x86-64.so.2 (0x0000000001021000)
[nico@hades ~/src/bm]$ /lib/x86_64-linux-gnu/libc.so.6 
GNU C Library (Debian GLIBC 2.24-11+deb9u4) stable release version 2.24, by Roland McGrath et al.
Copyright (C) 2016 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.
There is NO warranty; not even for MERCHANTABILITY or FITNESS FOR A
PARTICULAR PURPOSE.
Compiled by GNU CC version 6.3.0 20170516.
Available extensions:
        crypt add-on version 2.1 by Michael Glad and others
        GNU Libidn by Simon Josefsson
        Native POSIX Threads Library by Ulrich Drepper et al
        BIND-8.2.3-T5B
libc ABIs: UNIQUE IFUNC
For bug reporting instructions, please see:
<http://www.debian.org/Bugs/>.
[nico@hades ~/src/bm]$ 
```

--- a/nobuild.c
+++ b/nobuild.c
@@ -133,12 +133,20 @@ void wrappers_command(void)
     {
         CMD("cc", CFLAGS,
             "-c", "-fpic",
+#ifdef __FreeBSD__
+            // NOTE: Technically, this should be determined by pkg-config
+            "-I/usr/local/include",
+#endif
             "-I", PATH("src", "library"),
             "-o", PATH("build", "wrappers", "bm_sdl.o"),
             PATH("wrappers", "bm_sdl.c"));
 
         CMD("cc", CFLAGS,
             "-shared",
+#ifdef __FreeBSD__
+            // NOTE: See above
+            "-L/usr/local/lib",
+#endif
             "-o", PATH("build", "wrappers", "libbm_sdl.so"),
             PATH("build", "wrappers", "bm_sdl.o"),
             "-lSDL2");

--- a/src/library/path.c
+++ b/src/library/path.c
@@ -1,6 +1,8 @@
 #ifndef _WIN32
-#    define _DEFAULT_SOURCE
-#    define _POSIX_C_SOURCE 200112L
+#    ifdef __linux__
+#        define _DEFAULT_SOURCE
+#        define _POSIX_C_SOURCE 200112L
+#    endif
 #    include <sys/types.h>
 #    include <sys/stat.h>
 #    include <unistd.h>

